lib/Dialect/FIRRTL/Export/FIREmitter.cpp

```diff
@@ -1377,8 +1389,16 @@ void Emitter::emitType(Type type, bool includeConst) {
         if (type.getForceable())
           ps << "RW";
         ps << "Probe<";
+        ps.cbox(2, IndentStyle::Block);
+        ps.zerobreak();
         emitType(type.getType());
-        ps << ">";
+        if (auto layer = type.getLayer()) {
+          ps << ",";
+          ps.space();
+          emitSymbol(type.getLayer());
+        }
+        ps << BreakToken(0, -2) << ">";
```

Comment:
Awesome, thanks! Sorry for this being a thing and not abstracted away, but GJ :+1: .